### PR TITLE
Replace isNullLiteral with stream operators

### PR DESCRIPTION
**Pull request by raghavsethi**
As discussed [here](https://github.com/facebook/presto/pull/3709#discussion_r41449474)

### CODE REVIEW

**Comment by martint** [comment]
I'd leave it as isNullLiteral, invert the check inside the method and call `Iterables.none` instead. Having negatives in method names make the code harder to read.

**Comment by raghavsethi** [comment]
Iterables doesn't look it has a none method, and the presence of instanceOf Literal check complicates this a bit.

**Comment by martint** [comment]
Ah, I guess I was thinking of `Stream.noneMatch()`. We should update that code to use Java 8 streams instead of Guava `Iterables`, anyway.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java [hunk 1]
```diff
@@ -484,8 +484,8 @@ protected Object visitInPredicate(InPredicate node, Object context)
             // the analysis below. If the value is null, it means that we can't apply the HashSet
             // optimization
             if (!inListCache.containsKey(valueList)) {
-                if (Iterables.all(valueList.getValues(), ExpressionInterpreter::isNullLiteral)) {
-                    // if all elements are constant, create a set with them
+                if (valueList.getValues().stream().allMatch(i -> i instanceof Literal) &&
```
Comment (erichwang) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java]: Change
i -> i instanceof Literal
to
Literal.class::isInstance

File: presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java [hunk 2]
```diff
@@ -484,8 +484,8 @@ protected Object visitInPredicate(InPredicate node, Object context)
...
+                        valueList.getValues().stream().noneMatch(i -> i instanceof NullLiteral)) {
```
Comment (erichwang) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java]: Change
i -> i instanceof NullLiteral
to
NullLiteral.class::isInstance

**Comment by raghavsethi** [comment]
Thanks, made the replacements you suggested. Good to go?

**Comment by martint** [comment]
Looks good

**Pull request merged**